src/openvic-simulation/economy/ProductionType.cpp

```diff
@@ -29,96 +32,99 @@ bool ProductionType::parse_scripts(GameManager const& game_manager) {
 
 ProductionTypeManager::ProductionTypeManager() : rgo_owner_sprite { 0 } {}
 
-node_callback_t ProductionTypeManager::_expect_employed_pop(
-	GoodManager const& good_manager, PopManager const& pop_manager, callback_t<EmployedPop&&> cb
+node_callback_t ProductionTypeManager::_expect_job(
+	GoodManager const& good_manager, PopManager const& pop_manager, callback_t<Job&&> cb
 ) {
 
 	return [this, &good_manager, &pop_manager, cb](ast::NodeCPtr node) -> bool {
 		std::string_view pop_type;
-		EmployedPop::effect_t effect;
-		fixed_point_t effect_multiplier = 1, amount = 1;
+		Job::effect_t effect_type;
+		fixed_point_t effect_multiplier = 1, desired_workforce_share = 1;
 
-		using enum EmployedPop::effect_t;
-		static const string_map_t<EmployedPop::effect_t> effect_map = {
+		using enum Job::effect_t;
+		static const string_map_t<Job::effect_t> effect_map = {
 			{ "input", INPUT }, { "output", OUTPUT }, { "throughput", THROUGHPUT }
 		};
 
 		bool res = expect_dictionary_keys(
 			"poptype", ONE_EXACTLY, expect_identifier(assign_variable_callback(pop_type)),
-			"effect", ONE_EXACTLY, expect_identifier(expect_mapped_string(effect_map, assign_variable_callback(effect))),
+			"effect", ONE_EXACTLY, expect_identifier(expect_mapped_string(effect_map, assign_variable_callback(effect_type))),
 			"effect_multiplier", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(effect_multiplier)),
-			"amount", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(amount))
+			"amount", ZERO_OR_ONE, expect_fixed_point(assign_variable_callback(desired_workforce_share))
 		)(node);
 
 		const PopType* found_pop_type = pop_manager.get_pop_type_by_identifier(pop_type);
```

Comment:
```suggestion
		PopType const* found_pop_type = pop_manager.get_pop_type_by_identifier(pop_type);
		if (found_pop_type == nullptr && pop_type != "artisan") {
			Logger::error("Invalid pop type identifier \"", pop_type, "\"!");
			res = false;
		}
```
Even though it's checked later in `add_production_type`, there should still be an error message here if the pop type is invalid, which can include the actual invalid string.

Reply:
You can't do that. "artisan" is an invalid poptype. It's just that for artisanal production types, the owner and employees are ignored. If you add "artisan" in a factory or RGO, the game will crash. I tried to pass the production type to this method but it's not parsed yet.

The proper way to handle it is to first figure out if it is an artisanal production type or something else.
If it is, then warn about any owner or employee (apart from vanilla's artisan placeholder).
Else, error even on "artisan".